pkg/web/db-actions.ts

```diff
@@ -0,0 +1,78 @@
+let database: Database = {
+  notes: [],
+};
+interface Database {
+  notes: Note[];
+}
+export interface Note {
+  _id: string;
+  books: BookInterface[];
+}
+export interface BookInterface {
+  isbn: string;
+  title: string;
+  author: string;
+  editor: string;
+  price: number;
+  quantity: number;
+  note?: string;
+  date?: string;
+}
+
+export const newNote = (note: Note) => {
+  const currentDatabase = getDB();
+
+  const newDB = { notes: [...currentDatabase.notes, note] };
+
+  setDB(newDB);
+
+  const promise = new Promise<{ id: string; ok: boolean }>((resolve) => {
+    resolve({ id: note._id, ok: true });
+  });
+  return promise;
+};
+
+// get note
+export const getNote = (id: Note["_id"]) => {
+  const currentDatabase = getDB();
+
+  const note =
+    currentDatabase.notes.find((note) => note._id === id) || ({} as Note);
+
+  const promise = new Promise<Note>((resolve) => {
+    resolve(note);
+  });
+
+  return promise;
+};
+
+// set DB
+export const setDB = (newDB: Database) => {
+  database = newDB;
+};
+
+// get DB
+export const getDB = () => {
+  return database;
+};
+
+export const updateNote = (id: Note["_id"], newNote: Note) => {
+  const newDatabase = getDB();
+  const index = newDatabase.notes.findIndex(({ _id }) => _id === id);
+  newDatabase.notes.splice(index, 1);
+
+  setDB({ notes: [...newDatabase.notes, newNote] });
+
+  const promise = new Promise<{ id: string; ok: boolean }>((resolve) => {
+    resolve({ id, ok: true });
+  });
+  return promise;
+};
+
+export const deleteDatabase = () => {
```

Comment:
I don't think we need this.

For in-memory database (when the DB is a Javascript object) we'll let the garbage collector pick it up (or not - it's not the end of the world).
For PouchDB/CouchDB we can just leave the DB there (expecting the db server to be ephemeral and/or expecting the user to clean it up if they so desire).